01week/rockPaperScissors.js

```diff
@@ -3,49 +3,136 @@
 const assert = require('assert');
 const readline = require('readline');
 const rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
+    input: process.stdin,
+    output: process.stdout
 });
 
 
 function rockPaperScissors(hand1, hand2) {
 
-  // Write code here
+    // Write code here
+    //hand 1 enters their weapon, hand 2 enters their weapon - This is done
+
+    if (typeof(hand1) != 'string' && typeof(hand2) != 'string') {
+        return console.log('Invalid entry, please choose rock, paper, or scissors')
+    }
+    //making sure the hands entered are strings
+
+    const isRockPaperScissors = (hand) => {
+            if (hand != 'rock' && hand != 'paper' && hand != 'scissors') {
+                return true;
+            }
+        };
+        //writing a function to check if a hand is valid, then will use it for each hand
+    if (isRockPaperScissors(hand1)) {
+        return console.log('Invalid entry from player 1, please choose rock, paper, or scissors')
+    };
+    if (isRockPaperScissors(hand2)) {
+        return console.log('Invalid entry from player 2, please choose rock, paper, or scissors')
+    };
+    // if (hand1 != 'rock' && hand1 != 'paper' && hand1 != 'scissors') {
+    //     return console.log('Invalid entry from player 1, please choose rock, paper, or scissors')
+    // }
+    // //making sure hand1 is rock, paper, or scissors
+    //
+    // if (hand2 != 'rock' && hand2 != 'paper' && hand2 != 'scissors') {
+    //     return console.log('Invalid entry from player 2, please choose rock, paper, or scissors')
+    // }
+    // //making sure hand2 is rock, paper, or scissors
+
+    if (hand1 == hand2) {
+        return console.log("It's a tie!")
+    }
+    //checking for the tie first to narrow the cases
+
+    if (hand1 == 'rock' && hand2 == "paper") {
+        return console.log("Hand two wins!")
+    }
+
+    if (hand1 == 'rock' && hand2 == "scissors") {
+        return console.log("Hand one wins!")
+    }
+
+    if (hand1 == 'paper' && hand2 == "rock") {
+        return console.log("Hand one wins!")
+    }
+
+    if (hand1 == 'paper' && hand2 == "scissors") {
+        return console.log("Hand two wins!")
+    }
+
+    if (hand1 == 'scissors' && hand2 == "rock") {
+        return console.log("Hand two wins!")
+    }
+
+    if (hand1 == 'scissors' && hand2 == "paper") {
+        return console.log("Hand one wins!")
+    }
+    //covering the last 6 cases with individual if statements
```

Comment:
in this week we how to make decisions in Javascript. The although you statements are correct understand that it is taking a lot of processing time to check every if statement even if you have already found a true statement.